File: content/problems/1893-check-if-all-the-integers-in-a-range-are-covered/solutions/small-range-difference/Solution.java
class Solution {
    public boolean isCovered(int[][] ranges,int left,int right){
        int[] difference=new int[52];
        for(int[] range:ranges){
            difference[range[0]]++;
            difference[range[1]+1]--;
        }
        int coverage=0;
        for(int value=1;value<=right;value++){
            coverage+=difference[value];
            if(value>=left&&coverage==0)return false;
        }
        return true;
    }
}

File: content/problems/1893-check-if-all-the-integers-in-a-range-are-covered/solutions/small-range-difference/Solution.kt
class Solution {
    fun isCovered(ranges:Array<IntArray>,left:Int,right:Int):Boolean{
        val difference=IntArray(52);
        for(range in ranges){
            difference[range[0]]++;
            difference[range[1]+1]--
        };
        var coverage=0;
        for(value in 1..right){
            coverage+=difference[value];
            if(value>=left&&coverage==0)return false
        };
        return true
    }
}
